tensorflow_datasets/core/as_dataframe.py

```diff
@@ -82,9 +82,14 @@ def from_spec(
     elif sequence_rank > 1:
       repr_fn = feature.repr_html_ragged
 
-    def repr_fn_with_debug(val):  # Wrap repr_fn to add debug info
+    def repr_fn_wrapper(val):  # Wrap repr_fn to add debug info & fix html tags
       try:
-        return repr_fn(val)
+        html_repr =  repr_fn(val)
+        # Replace '<' and '>' characters in the html with '&lt;' and '&gt;'
+        # so that they are not confused as html tags when rendering
+        # Inspired from pandas _repr_html_
+        html_repr = html_repr.replace('<', r'&lt;', 1).replace('>', r'&gt;', 1)
```

Comment:
This is not the right place to escape this, as `repr_fn` might also returns actual HTML code (e.g. to display images, video,...)

Instead, we should make the update in `def _repr_html()` in `core/features/feature.py`